server/src/main/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorFactory.java

```diff
@@ -29,7 +30,12 @@ class MaxAggregatorFactory extends ValuesSourceAggregatorFactory {
     static void registerAggregators(ValuesSourceRegistry.Builder builder) {
         builder.register(
             MaxAggregationBuilder.REGISTRY_KEY,
-            List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.DATE, CoreValuesSourceType.BOOLEAN),
+            List.of(
+                CoreValuesSourceType.NUMERIC,
+                CoreValuesSourceType.DATE,
+                CoreValuesSourceType.BOOLEAN,
+                TimeSeriesValuesSourceType.COUNTER
```

Comment:
Should Gauge also be supported here?

Reply:
There is no custom value type for gauge today. Only for counters. We added this for the rate aggregation, so that we trigger a different implementation that is able to detect resets.